docker-app/qfieldcloud/core/views/users_views.py

```diff
@@ -146,3 +168,235 @@ class ListUserOrganizationsView(generics.ListAPIView):
+        team_name = f'@{organization_name}/{team_name}'
+        team = get_object_or_404(Team, team_organization=organization, username=team_name)
+
+        new_team_name = request.data.get('username')
+        if not new_team_name:
+            return Response({"error": "New team name is required."}, status=status.HTTP_400_BAD_REQUEST)
+
+        # Check if the new name is already taken
+        if Team.objects.filter(team_organization=organization, username=new_team_name).exists():
+            return Response({"error": "Team with this name already exists."}, status=status.HTTP_400_BAD_REQUEST)
+
+        # Update the team name
+        new_team_name = f'@{organization_name}/{new_team_name}'
+        team.username = new_team_name
+        team.save()
+
+
+        serializer = TeamSerializer(team)
+        return Response(serializer.data, status=status.HTTP_200_OK)
+        # return Response({"username": team.username}, status=status.HTTP_200_OK)
+
+    def delete(self, request, organization_name, team_name):
+        """
+        DELETE /organizations/<str:organization_name>/team/<str:team_name>/
+        Delete a team within the organization.
+        """
+        organization = get_object_or_404(Organization, username=organization_name)
+        team_name = f'@{organization_name}/{team_name}'
+        team = get_object_or_404(Team, team_organization=organization, username=team_name)
+
+        # Delete the team
+        team.delete()
+        serializer = TeamDeleteSerializer(data={'message': "Team deleted successfully"})
+        return Response(serializer.data, status=status.HTTP_204_NO_CONTENT)
+
+        # return Response({"message": "Team deleted successfully"}, status=status.HTTP_204_NO_CONTENT)
+
+
+
+class TeamListCreateView(APIView):
+
+    def get(self, request, organization_name):
+        """
+        Handle GET request: Retrieve all teams under a specific organization --> organizations/<str:organizationr_name>/team/
+        """
+        # Get the organization by its name
+        organization = get_object_or_404(Organization, username=organization_name)
+
+        teams = querysets_utils.get_organization_teams(organization)
+
+        serializer = TeamListSerializer(teams, many=True)
+        return Response(serializer.data, status=status.HTTP_200_OK)
+
+        # team_list = [
+        #     {
+        #         "Team Name": team.username,
+        #         "Actual name:": team.teamname
+        #     }
+        #     for team in teams
+        # ]
+
+        # return Response(team_list, status=status.HTTP_200_OK)
+
+    def post(self, request, organization_name):
+        """
+        Handle POST request: Create a new team under a specific organization.
+        """
+        # Get the organization by its name
+        organization = get_object_or_404(Organization, username=organization_name)
+
+        # Extract the 'username' field from the request data
+        team_name = request.data.get('username')
```

Comment:
I think these are quite obvious, so you can drop the comments.
```suggestion
        organization = get_object_or_404(Organization, username=organization_name)
        team_name = request.data.get('username')
```

Reply:
Accepted and Solved!